src/components/side-panel/side-panel.tsx

```diff
@@ -30,10 +30,22 @@ export const SidePanel = ({ offset, onSubmit }: IProps) => {
     const { showToast } = useDisplay();
     const methods = useForm({
         mode: "onTouched",
+        defaultValues: {
+            type: "",
+            label: "",
+            required: true,
+            requiredErrorMsg: "",
+            id: "",
+            placeholder: "",
+        },
```

Comment:
can we set the default values in each `Controller` instead? because the fields rendered differ between element types

Reply:
From RHF [docs](https://react-hook-form.com/docs/usecontroller), in order to use defaultValue in `Controller`, we need to define `defaultValue` in `useForm`. 

We can remove the default value for the rest of the fields actually, but for `requiredErrorMsg`, if we don't, when it remounts, `defaultValue` for `requiredErrorMsg` is set to `undefined` in the form. Having `defaultValue` set in the `Controller` will only set the value of the field, but the form will still expect `undefined` in order to be clean.

Reply:
ok fair point, let's go with this then